### PR TITLE
OP-184 | Convert to function component

### PR DESCRIPTION
**Pull request by almeida1492**
Convert components under `PatientDatabase` into function components in order to update the application code. I point out that it's a work that has to be propagated gradually throughout the project, as it has an old-fashioned way of writing React.

**Resolves:** [OP-184](https://openhospital.atlassian.net/browse/OP-184)

### CODE REVIEW

File: src/components/patientsDatabase/PatientsDatabase.tsx [hunk 1]
```diff
@@ -34,200 +32,182 @@ import MergeIcon from "@material-ui/icons//LibraryBooks";
...
 import CancelIcon from "@material-ui/icons/Cancel";
 
-// constants
-import { PATH_NEW_PATIENT } from "../../helpers/constants";
-
-export interface LocalProps extends WithStyles<typeof styles> {
+interface IOwnProps extends WithStyles<typeof styles> {
   classes: any;
   classNames: any;
+  isDeleteDialogOpen: boolean;
```
Comment (alefalezza) [inline src/components/patientsDatabase/PatientsDatabase.tsx]: Do you use it? I think it should be removed

File: src/components/patientsDatabase/PatientsListItem.tsx [hunk 2]
```diff
@@ -20,94 +20,95 @@ import Avatar from "@material-ui/core/Avatar";
...
-                                </Grid>
-                            </Grid>
-                        </Grid>
-                    </MaterialCardActionAreaRouter>
-                </Paper>
-            </Grid>
-        )
-    }
+interface Props {
+  classes: any;
```
Comment (alefalezza) [inline src/components/patientsDatabase/PatientsListItem.tsx]:
```suggestion
interface Props extends WithStyles<typeof styles> {
```

File: src/components/patientsDatabase/PatientActivityContainer.tsx [hunk 3]
```diff
@@ -16,167 +16,165 @@ import PatientExamination from "./PatientExamination";
...
+  PATH_PATIENT_NEW_VACCINATION,
+  PATH_OPD,
+  PATH_NEW_OPD,
+} from "../../helpers/constants";
+import { Patient } from "types/patients";
+import { AppState } from "reducers";
 
 export interface LocalProps extends WithStyles<typeof styles> {
-    classes: any
+  classes: any;
```
Comment (alefalezza) [inline src/components/patientsDatabase/PatientActivityContainer.tsx]:
```suggestion
```
Classe should be removed as it's given by WhitStyles

File: src/components/patientsDatabase/PatientActivityContainer.tsx [hunk 4]
```diff
@@ -16,167 +16,165 @@ import PatientExamination from "./PatientExamination";
...
-        } = this.props
-
-        if (location.patient) {
-            putPatientInStore(location.patient)
-        } else {
-            getPatientInServer(match.params.patientId)
-        }
+  componentDidMount() {
+    const { match, putPatientInStore, getPatientInServer, location } = this.props;
+    console.log(location, match);
```
Comment (alefalezza) [inline src/components/patientsDatabase/PatientActivityContainer.tsx]:
```suggestion
```
Remove consol.log

File: src/components/utils/LinkHelper.tsx [hunk 5]
```diff
@@ -1,27 +1,23 @@
...
+import { LinkProps, NavLinkProps } from "react-router-dom";
+import { FunctionComponent } from "react";
 
-
-export const MaterialLinkRouter = (props: LinkProps) => {
-  return (
-    <Link {...props} />
-  );
+interface IMaterialLinkRouter extends LinkProps {
+  component: any;
```
Comment (alefalezza) [inline src/components/utils/LinkHelper.tsx]: I'm missing something, why this prop?

Comment (almeida1492) [reply src/components/utils/LinkHelper.tsx]: It's an old thing. I don't know why it's there too. I think the original idea was to have a customizable MaterialLinkRouter.

**Comment by alefalezza** [comment]
@mwithi  If you agree, can be merged

**Comment by almeida1492** [comment]
@mwithi Description added.

**Comment by xrmx** [comment]
I see there are typescript warnings for these files, don't know if are old or new but worth fixing

**Comment by almeida1492** [comment]
@xrmx There are typescript errors throughout the whole project because there was almost no ts code written before. Sometimes fixing them demands a load of work because they can go deep. Each one of these you've seen is old. I tried to fix the ones directly related to the stuff I was refactoring for not losing focus on the task's subject. The ones you've seen in these files are mainly linked to the `LinkHelper`, and I'm not even sure if it's going to remain in the code.

File: src/components/patientsDatabase/PatientsDatabase.tsx [hunk 6]
```diff
@@ -34,200 +32,181 @@ import MergeIcon from "@material-ui/icons//LibraryBooks";
...
-              <CircularProgress className={classes.progress} color="secondary" style={{ margin: "20px auto" }} />
-            ) : (
-              patients.map(patient => <PatientsListItem key={patient.id} patient={patient} />)
-            )}
-          </Grid>
-          <Grid item xs={12} sm={2} className={classes.loadMoreContainer}>
+
+            <Button
+              color="inherit"
+              onClick={() => setShowDeletePatientDialog(true)}
```
Comment (mwithi) [inline src/components/patientsDatabase/PatientsDatabase.tsx:86]: All this part will be refactored, so for now it will be merged as is

**Pull request merged**